nav2_costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -86,31 +70,40 @@ Costmap2DROS::Costmap2DROS(const std::string & name, tf2_ros::Buffer & tf)
 {
   tf2::toMsg(tf2::Transform::getIdentity(), old_pose_.pose);
 
-  private_nh_ = rclcpp::Node::make_shared(name_);
+  node_ = std::shared_ptr<rclcpp::Node>(this, [](rclcpp::Node *) {});
+
+  // Set Parameters if not set 
+  this->set_parameter_if_not_set("transform_tolerance",0.3);
```

Comment:
Why are you setting the parameter instead of just calling get_parameter_or?

Reply:
These parameters are all dynamic parameters, thus I'm setting them such that the `dynamic_params_client_` will be able to initialize them in the cached map from the node.